src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,110 @@
-import React from 'react';
+import cn from 'classnames';
+import React, { useState } from 'react';
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  selectedPost: Post | null;
+  setCommentsOfPost: (comments: Comment[] | null
+  | ((prevVar: Comment[] | null) => Comment[] | null)) => void;
+  setIsErrorShowing: (value: boolean) => void;
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  setCommentsOfPost,
+  setIsErrorShowing,
+}) => {
+  const [formInputs, setFormInputs] = useState({
+    author: '',
+    email: '',
+    text: '',
+  });
+  const [formErrors, setFormErrors] = useState({
+    author: false,
+    email: false,
+    text: false,
+  });
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const handleChangeInput = (
+    value: string,
+    field: string,
+  ): void => {
+    setFormInputs(prevForm => ({
+      ...prevForm,
+      [field]: value,
+    }));
+
+    setFormErrors(prevErrors => ({
+      ...prevErrors,
+      [field]: false,
+    }));
+  };
+
+  const handleReset = () => {
+    setFormInputs({
+      author: '',
+      email: '',
+      text: '',
+    });
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsErrorShowing(false);
+
+    const fields = Object.keys(formInputs);
+    let hasError = false;
+
+    fields.forEach((key) => {
+      if (!formInputs[key as keyof typeof formInputs]) {
+        setFormErrors(prevErrors => ({
+          ...prevErrors,
+          [key]: true,
+        }));
+
+        hasError = true;
+      }
+    });
+
+    if (hasError) {
+      return;
+    }
+
+    const newComment = {
+      postId: selectedPost?.id,
+      name: formInputs.author,
+      email: formInputs.email,
+      body: formInputs.text,
+    };
+
+    setIsSubmitting(true);
+    client.post<Comment>('/comments', newComment)
+      .then((comment) => {
+        setCommentsOfPost(prevComments => {
+          if (prevComments !== null) {
```

Comment:
```suggestion
          if (prevComments) {
```
You don't need to strict compare with null just check if prevComments exists